examples/weather/dlwp_healpix/trainer.py

```diff
@@ -0,0 +1,480 @@
+                                output = self.model(inputs)
+                                validation_stats[0] += self.criterion(prediction=output, target=target) * bsize
+                                # save per variable loss
+                                eval_losses = self.criterion(output, target, average_channels=False)
+                                for v_idx, v_name in enumerate(self.output_variables):
+                                    validation_stats[1+v_idx] += eval_losses[v_idx] * bsize
+                        else:
+                            output = self.model(inputs)
+                            validation_stats[0] += self.criterion(prediction=output, target=target) * bsize
+                            eval_losses = self.criterion(output, target, average_channels=False)
+                            for v_idx, v_name in enumerate(self.output_variables):
+                                validation_stats[1+v_idx] += eval_losses[v_idx] * bsize
+
+                    pbar.set_postfix({"Loss": (validation_stats[0]/validation_stats[-1]).item()})
+
+                    # increment sample counter
+                    validation_stats[-1] += bsize
+
+                if torch.distributed.is_initialized():
+                    torch.distributed.all_reduce(validation_stats)
+
+                validation_error = (validation_stats[0] / validation_stats[-1]).item()
+
+                # Record error per variable
+                validation_errors = []
+                for v_idx, v_name in enumerate(self.output_variables):
+                    validation_errors.append((validation_stats[1+v_idx]/validation_stats[-1]).item())
+
+                # Track validation improvement to later check early stopping criterion
+                if validation_error < best_validation_error:
+                    best_validation_error = validation_error
+                    epochs_since_improved = 0
+                else:
+                    epochs_since_improved += 1
+
+
+            torch.cuda.nvtx.range_pop()
+
+            # Logging and checkpoint saving
+            if self.dist.rank == 0:
+                if self.lr_scheduler is not None:
+                    self.writer.add_scalar(tag="learning_rate", scalar_value=self.optimizer.param_groups[0]['lr'],
+                                           global_step=iteration)
+                self.writer.add_scalar(tag="val_loss", scalar_value=validation_error, global_step=iteration)
+                
+                # Per-variable loss
+                for v_idx, v_name in enumerate(self.output_variables):
+                    self.writer.add_scalar(tag=f"val_loss/{v_name}", scalar_value=validation_errors[v_idx],
+                                           global_step=iteration)
+
+                # Write model checkpoint to file, using a separate thread
+                self.logger0.info("Writing checkpoint")
+                thread = threading.Thread(
```

Comment:
Should switch to the modulus checkpointing format as discussed in the meeting last week: https://github.com/NVIDIA/modulus/blob/main/modulus/launch/utils/checkpoint.py#L174

Reply:
Updated to modulus checkpointing